R/utilsFiles.R

```diff
@@ -376,10 +385,49 @@ xlsxFindReplace <-
     # and a message is displayed with number of changes
     catret("File ", xlsxName, " ", result, " Changes ")
     result
+    output <- list(result)
+    names(output) <- xlsxName    
+    return(output)
   }
 
 
+#' Get list of all filepaths and subsidiaries within a directory
+#'
+#' This function starts with a given folder and returns the filepaths of all files
+#' in all the folders. You can specify specific folders to ignore, and specific filetypes
+#' to find. 
+#' 
+#' @param directory The filepath where you want to search.
+#' @param filetypes The types of file you want to list (optional) (input is eg. c("R", "Rmd").
+#' @param ignore.these The names of any folders you wish to ignore.
 
+#'
+#' @return Returns list of complete filepaths
+#' @export
+#' @examples
+#' \dontrun{
+#' allpaths <- getAllFilepaths(getwd(),filetypes = c("R","Rmd","txt"), ignore.these = c(".git", ".gitignore",".Rproj.user"))
+#' }
+getAllFilepaths <- function(directory, filetypes = NULL, ignore.these=NULL){
```

Comment:
Also works great for me in my tested directly. Are we intending to combine this with find and replace to bulk change any mention?

Reply:
For finding locations for / making bulk changes, this combo works to get a data frame that can then query to identify all the files with the old filepath segment (and change at same time if needed).

allfiles <- getAllFilepaths(getwd(),filetypes = c("R","Rmd","txt"), ignore.these = c(".git", ".gitignore",".Rproj.user"))
test <- filesFindReplace(allfiles ,"[oldfilepathsegment]","[newfilepathsegment]", listonly = T)
res <- data.frame("filename" = names(test)
                  ,"changes" = test)

Reply:
also thanks for reviewing :D

Reply:
You should manage to never have path in your source ;) 
But this is probably for old sources ;)